packages/chat-component-bindings/src/messageThreadSelector.ts

```diff
@@ -136,22 +161,46 @@ export const messageThreadSelector: MessageThreadSelector = createSelector(
             message.clientMessageId !== undefined
         )
         .filter((message) => message.content && message.content.message !== '') // TODO: deal with deleted message and remove
-        .map((message) =>
-          memoizedFn(
+        .map((message) => {
+          /** logic: Looking at message A, how do we know it's read number?
+           * Assumption: if user read the latest message, user has read all messages before that
+           * ReadReceipt behaviour: read receipt is only sent to the last message
+           * if participant read a message that is sent later than message A, then the participant has read message A
+           * how do we check if the message is sent later than message A?
+           * we compare if the readon timestamp is later than the message A sent time
+           * if last read message id is not equal to message A's id, check the read on time stamp.
+           * if the last read message is read after the message A is sent, then user should have read message A as well */
```

Comment:
<3 this comment!
This is the sort of caclucation that lends itself very well to unit tests (to easily have a unit test for just this this should be its own fn `calculateReadReceipts(message, latestReadReceiptPerSender)` -- maybe could be added as part of https://skype.visualstudio.com/SPOOL/_workitems/edit/2762970

Reply:
Yeah I have a task create for adding unit tests